src/print_diff.ml

```diff
@@ -24,14 +24,17 @@ let print ?(skip_trailing_cr=Sys.win32) path1 path2 =
       ]
   in
   let normal_diff () =
-    match Bin.which ~path:(Env.path Env.initial) "diff" with
+    let which bin = Option.map (Bin.which ~path:(Env.path Env.initial) bin) ~f:(fun path -> path, bin) in
+    let (|||) a b = if a = None then b else a in
+    match which "git" ||| which "diff" with
     | None -> fallback ()
-    | Some prog ->
+    | Some (prog, bin) ->
```

Comment:
The code flow is a bit complicated. I would do something simpler:
```ocaml
let path, args =
  let which prog = Bin.which ~path:(Env.path Env.initial) prog in
  match which "git" with
  | Some path ->
      path, [ "diff"; "--no-index"; "--color=always"; "-u" ]
  | None ->
      begin match which "diff" with
      | Some path ->
          path, [ "-u" ]
      | None ->
          fallback ()
      end
in
let args = if skip_trailing_cr then args @ [ "--strip-trailing-cr" ] in
let args = args @ [ file1; file2 ] in
let* () = Processs.run ~dir ~env:Env.initial Strict path args in
...
```

Also, one should only pass the `--color=always` flag only if `Ansi_color.stdout_supports_color` is `true`.

Reply:
@nojb, actually we should pass `--color=always` all the time. Dune will strip the colors if its own output is a terminal. In a way, Dune moves the color settings from applications it runs to itself.

The other changes you suggest look good to me.

Reply:
Ah yes, indeed!